File: src-tauri/src/tasks/task_manage.rs
// src/tasks/task_manage.rs

use crate::entities::task::Task; // 引入 Task 结构体
use crate::tasks::itask_manage::{IPersistence, ITaskManage}; // 引入 ITaskManage trait

pub struct TaskManager {
    pub persistence: Box<dyn IPersistence>,
}

impl TaskManager {
    pub fn new(persistence: Box<dyn IPersistence>) -> Self {
        TaskManager { persistence }
    }
}

// 为 TaskManager 实现 ITaskManage trait
impl ITaskManage for TaskManager {
    fn crate_task(&mut self, task: Task) -> Result<Task, String> {
        let mut task = task;
        let task_id = Task::get_id();
        task.id = Some(task_id);
        self.persistence.save_task(&task)?;
        Ok(task)
    }

    fn modify_task(&mut self, task: Task) -> Result<Task, String> {
        return Err("开发中 todo!()".to_string());
    }

    fn query_task_by_time(&self, start_time: u64, end_time: u64) -> Result<Vec<Task>, String> {
        return Err("开发中 todo!()".to_string());
    }

    fn delete_task(&mut self, task_id: String) -> Result<Task, String> {
        return Err("开发中 todo!()".to_string());
    }

    fn complete_task(&mut self, task_id: String) -> Result<Task, String> {
        return Err("开发中 todo!()".to_string());
    }
}

File: src/service/TaskManage.ts
import { Task } from "@/entities/taskEntities";



/**
 * 任务管理接口
 */
/**
 * 任务管理器抽象类，定义了任务管理的基本操作。
 */
export abstract class TaskManager {
     /**
      * 创建一个新任务。
      * @param task 要创建的任务对象。
      * @returns 返回创建的任务对象的 Promise。
      */
     abstract createTask(task: Task): Promise<Task>;

     /**
      * 修改一个已有的任务。
      * @param task 要修改的任务对象。
      * @returns 返回修改后的任务对象的 Promise。
      */
     abstract modifyTask(task: Task): Promise<Task>;

     /**
      * 根据时间范围查询任务。
      * @param startTime 查询的开始时间（时间戳）。
      * @param endTime 查询的结束时间（时间戳）。
      * @returns 返回任务对象数组的 Promise。
      */
     abstract queryTaskByTime(startTime: number, endTime: number): Promise<Task[]>;

     /**
      * 删除一个任务。
      * @param taskId 要删除的任务的 ID。
      * @returns 返回删除的任务对象的 Promise。
      */
     abstract deleteTask(taskId: string): Promise<Task>;

     /**
      * 完成一个任务。
      * @param taskId 要完成的任务的 ID。
      * @returns 返回完成的任务对象的 Promise。
      */
     abstract completeTask(taskId: string): Promise<Task>;
}
